src/node/peer/messages.rs

```diff
@@ -139,6 +142,27 @@ impl Peer {
         Ok(())
     }
 
+    pub async fn handle_bulk_pull(
+        &mut self,
+        _header: &Header,
+        _bulk_pull: BulkPull,
+    ) -> anyhow::Result<()> {
+        // The rest of this connection will be a bunch of blocks without any headers.
+        if matches!(self.bootstrap_state, FrontierStream) {
+            panic!("Invalid bootstrap state transition FrontierStream => BulkPull");
```

Comment:
Probably planned, but just suggesting this be an `Err` because a peer can trigger this, right?

Reply:
Not sure, I'm moving stuff around right now. Could you elaborate anyways? I used panic because I thought this should be an irrecoverable programming error, the transition should not be even possible.